File: Solution2582.cs
/*
submitted at Jul 06, 2024 15:02

Runtime 22 ms, Beats 47.83%
Memory 26.62 MB, Beats 80.43%
*/

public class Solution {
    public int PassThePillow(int n, int time) {
        int mod = 2 * (n - 1);
        int curr = time % mod;
        return 1 + (curr < n ? curr : mod-curr);
    }
}

File: Solution2582.rs
/*
submitted at Jul 06, 2024 15:10
Time 0ms Beats 100.00%
Memory 2.09MB Beats 50.00%
*/

impl Solution {
    pub fn pass_the_pillow(n: i32, time: i32) -> i32 {
        let m = 2 * (n - 1);
        let curr = time % m;
        
        if curr < n 
            {curr+1}
        else
            {m-curr+1}
    }
}
